odxtools/parameters/lengthkeyparameter.py

```diff
@@ -44,13 +44,15 @@ def is_optional(self):
         return True
 
     def encode_into_pdu(self, encode_state: EncodeState) -> bytes:
-        physical_value = encode_state.parameter_values.get(self.short_name,)
+        physical_value = encode_state.parameter_values.get(self.short_name)
 
         if physical_value is None:
             raise TypeError(f"A value for the length key '{self.short_name}'"
                             f" must be specified.")
 
         # Set the value of the length key in the length key dict.
-        encode_state.length_keys[self.odx_id] = physical_value
+        encode_state.length_keys[self.short_name] = physical_value
```

Comment:
this will cause collision if two lengths in different structs use a common generic name like "len", "length", "size"...

Reply:
yes, but I assume that short names must be unique within each structure (as far as I remember, there are SNREFs to parameters somewhere...)